lib/vehicle_factory.rb

```diff
@@ -0,0 +1,27 @@
+class VehicleFactory
+
+    #takes formatted data and puts into var list
+  def create_vehicles(registrations)
+      vehicles = format(registrations)
+      vehicles.map do |vehicle|
+          Vehicle.new(vehicle)
+      end
+  end
+
+  # takes data and formats to fit data taken for vehicle class
+  def format(registrations)
+      formatted = []
+      [registrations].flatten.each do |registration|
+          newformat = {}
+          newformat[:vin] = registration[:vin_1_10]
+          newformat[:year] = registration[:model_year]
+          newformat[:make] = registration[:make]
+          newformat[:model] = registration[:model]
+          newformat[:engine] = :ev
+          formatted << newformat
+      end
+      formatted
+  end
+
+  #end class
+end
```

Comment:
This looks really good, and again, I like that you're getting the practice with creating your own helper methods. We could maybe slim this down a little bit. If we did do one method, we could do something like: 

```ruby
  def create_vehicles(registrations)
     registrants.map do |vehicle|
        Vehicle.new({
          engine: :ev,
          make: vehicle[:make],
          model: vehicle[:model],
          vin: vehicle[:vin_1_10],
          year: vehicle[:model_year]
        })
 end
```